app/services/facebook_oidc_auth_service.rb

```diff
@@ -0,0 +1,92 @@
+# frozen_string_literal: true
+
+require "jwt"
+require "net/http"
+
+class FacebookOidcAuthService < BaseAuthService
+  class << self
+    private
+
+    def primary_key
+      :facebook_user_id
+    end
+
+    def expected_fields
+      %w[user_id]
+    end
+
+    def decode_token(id_token)
+      jwks = fetch_facebook_jwks
+      jwt_payload = JWT.decode(id_token, nil, true, {
+        algorithms: ["RS256"],
+        jwks: jwks,
+        verify_aud: true,
+        aud: ENV.fetch("FACEBOOK_APP_ID"),
+        verify_iss: true,
+        iss: "https://www.facebook.com"
+      })[0]
```

Comment:
[Nice-to-have] JWT validation options duplicated in tests. Evidence: Similar JWT.decode options in lines 19-27. Impact: Test maintenance burden. Suggestion: Extract to shared configuration. [Category: File]

Reply:
Closing; I believe we don't care if tests are WET